test/APRCalculator/MacroFactor.test.ts

```diff
@@ -79,12 +69,12 @@ export function RunMacroFactorTests(): void {
       );
       const currentMacroFactor = await aprCalculator.getMacroFactor();
       expect(currentMacroFactor).to.equal(INITIAL_DEFAULT_MACRO_FACTOR);
-      expect(await aprCalculator.disabledMacro()).to.be.true;
+      expect(await aprCalculator.disabledBonusesUpdates()).to.be.true;
       expect(await aprCalculator.smaSlowSum()).to.equal(oldSMASlowSum);
       expect(await aprCalculator.smaFastSum()).to.equal(oldSMAFastSum);
     });
 
-    it("should disable gard and enable macro factor updates after calling the function again", async function () {
+    it("should disable guard and enable macro factor updates after calling the function again", async function () {
```

Comment:
Is there a test about what would happen if a price update is made when the functionality is disabled. What will be the price when the functionality is re-enabled?

Reply:
There were tests for the event to determine if it emits or not. Now I added tests to check when we guard and update prices, that the avrageGain/Loss is not changed and when we disable the guard it does it is changed with the newest only update prices that were triggered after the guard is disabled. Also, added tests to check we calculate the RSI correctly!